cpp/skyweaver/src/PipelineConfig.cpp

```diff
@@ -157,6 +157,47 @@ DedispersionPlan const& PipelineConfig::ddplan() const
     return _ddplan;
 }
 
+std::size_t PipelineConfig::convertMemorySize(const std::string& str) const {
+    std::size_t lastCharPos = str.find_last_not_of("0123456789");
+    std::string numberPart = str.substr(0, lastCharPos);
+    std::string unitPart = str.substr(lastCharPos);
+
+    std::size_t number = std::stoull(numberPart);
+
+    if (unitPart.empty())
+        return number;
+    else if (unitPart == "K" || unitPart == "k")
+        return number * 1024;
+    else if (unitPart == "M" || unitPart == "m")
+        return number * 1024 * 1024;
+    else if (unitPart == "G" || unitPart == "g")
+        return number * 1024 * 1024 * 1024;
+    else
+        throw std::runtime_error("Invalid unit!");
+}
+
+void PipelineConfig::configure_wait(std::string argument)
```

Comment:
There is no error handling on this function. std::stoi, std::stoull, etc. do not throw error on out of range conversions. You can either check for ERANGE errors from errno after the calls or you can check if the return values are equal to their type's max, e.g. `_wait.sleep_time == std::numeric_limits<unsigned long long>::max();`